lib/openstudio-standards/hvac_sizing/Siz.HeatingCoolingFuels.rb

```diff
@@ -196,6 +196,358 @@ def coil_heating_fuels(heating_coil)
+        heat_types << self.coil_heat_type(component)  
+      when 'OS_Coil_Heating_WaterToAirHeatPump_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_Heating_WaterToAirHeatPump_VariableSpeed_EquationFit'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_AirToWaterHeatPump'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Coil_WaterHeating_Desuperheater'
+        heat_types << self.coil_heat_type(component)
+      when 'OS_Node', 'OS_Fan_ConstantVolume', 'OS_Fan_VariableVolume', 'OS_AirLoopHVAC_OutdoorAirSystem'
+        # To avoid extraneous debug messages  
+      else
+        #OpenStudio::logFree(OpenStudio::Debug, 'openstudio.sizing.Model', "No heating fuel types found for #{obj_type}")
+      end
+    end    
+
+    has_electric = false
+    has_fuel = false
+    has_district = false
+    has_solar = false
+    has_hp = false
+    has_mixed = false
+    if heat_types.size > 0
+      heat_types.sort.each do |heat_type|
+         case heat_type
+        when 'Electricity', 'Electric'
+          has_electric = true
+        when 'DistrictHeating', 'District'
+          has_district = true
+        when 'SolarEnergy', 'Solar'
+          has_solar = true
+        when 'NaturalGas', 'Fuel', 'PropaneGas', 'Coal', 'Diesel'
+          has_fuel = true
+        when 'FuelOilNo1', 'FuelOilNo2'
+          has_fuel = true
+        when 'HeatPump'
+          has_hp = true
+        when 'Mixed'
+          has_mixed = true
+        end
+      end
+    end
+
+    # Choose only one heat type, or Mixed if there are multiple
+    heat_type = ''
+    if has_mixed == true
+      heat_type = 'Mixed'
+    end
+    if has_fuel == true && heat_type != 'Fuel'
+      if heat_type == ''
+        heat_type = 'Fuel'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_district == true && heat_type != 'District'
+      if heat_type == ''
+        heat_type = 'District'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_electric == true && heat_type != 'Electric'
+      if heat_type == ''
+        heat_type = 'Electric'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_solar == true && heat_type != 'Solar'
+      if heat_type == ''
+        heat_type = 'Solar'
+      else
+        heat_type = 'Mixed'
+      end
+    end
+    if has_hp == true && heat_type != 'HeatPump'
+      # Assume any additional heat types are supplemental
+      heat_type = 'HeatPump'
+    end
+
+    return heat_type
+
+  end
+
+  # Get primary type of heat for one air loop for Appendix G system type selections
+  # 
+  # @param air_loop [object] 
+  # @return string description of heat type: Fuel, Electric, District, HeatPump, Solar, Mixed
+  def zone_equipment_heat_type(zone)
+    fuels = []
+    heat_type = ''
+    # Get the heating fuels for all zone HVAC equipment
+    zone.equipment.each do |equipment|
```

Comment:
should be zone.equipment.sort.each

Reply:
@mdahlhausen: I've noticed sometimes sort is included in object loops and sometimes it is not. Should it always be used? What is the purpose?

Reply:
yes, the sort should always be used.  Otherwise you can get some non-deterministic behavior.  Tough lesson from experience on that.